kiwi/boot/image/__init__.py

```diff
@@ -28,14 +28,16 @@ class BootImage(object):
     """
         BootImge factory
     """
-    def __new__(self, xml_state, target_dir, root_dir=None):
+    def __new__(self, xml_state, target_dir, root_dir=None, signing_keys=None):
         initrd_system = xml_state.build_type.get_initrd_system()
         if not initrd_system:
             initrd_system = 'kiwi'
         if initrd_system == 'kiwi':
-            return BootImageKiwi(xml_state, target_dir)
+            return BootImageKiwi(
+                xml_state, target_dir, signing_keys=signing_keys
+            )
         elif initrd_system == 'dracut':
-            return BootImageDracut(xml_state, target_dir, root_dir)
+            return BootImageDracut(xml_state, target_dir)
```

Comment:
Hmm, this looks wrong to me. In case of a dracut initrd system we _have_ to use the root_dir to operate and don't let the initrd creation process create a new and empty tmpdir. That's because dracut has to have access to the root system data it should boot later on. I also wonder how that change depend on the signing key implementation ?